src/Workspaces/Core/Portable/Diagnostics/InternalDiagnosticsOptions.cs

```diff
@@ -6,18 +6,39 @@ namespace Microsoft.CodeAnalysis.Diagnostics
-        public static readonly Option<bool> UseCompilationEndCodeFixHeuristic = new Option<bool>(OptionName, "Enable Compilation End Code Fix With Heuristic", defaultValue: true);
-        public static readonly Option<bool> BuildErrorIsTheGod = new Option<bool>(OptionName, "Make build errors to take over everything", defaultValue: false);
-        public static readonly Option<bool> ClearLiveErrorsForProjectBuilt = new Option<bool>(OptionName, "Clear all live errors of projects that got built", defaultValue: false);
-        public static readonly Option<bool> PreferLiveErrorsOnOpenedFiles = new Option<bool>(OptionName, "Live errors will be preferred over errors from build on opened files from same analyzer", defaultValue: true);
-        public static readonly Option<bool> PreferBuildErrorsOverLiveErrors = new Option<bool>(OptionName, "Errors from build will be preferred over live errors from same analyzer", defaultValue: true);
-        public static readonly Option<bool> PutCustomTypeInBingSearch = new Option<bool>(OptionName, "PutCustomTypeInBingSearch", defaultValue: true);
-        public static readonly Option<bool> CrashOnAnalyzerException = new Option<bool>(OptionName, "CrashOnAnalyzerException", defaultValue: false);
-        public static readonly Option<bool> ProcessHiddenDiagnostics = new Option<bool>(OptionName, "Process Hidden Diagnostics", defaultValue: false);
+        private const string LocalRegistryPath = @"Roslyn\Internal\Diagnostics\";
+
+        public static readonly Option<bool> BlueSquiggleForBuildDiagnostic = new Option<bool>(nameof(InternalDiagnosticsOptions), "Blue Squiggle For Build Diagnostic", defaultValue: false,
+            serializations: new LocalUserProfileSerialization(LocalRegistryPath + "Blue Squiggle For Build Diagnostic"));
+
+        public static readonly Option<bool> UseDiagnosticEngineV2 = new Option<bool>(nameof(InternalDiagnosticsOptions), "Use Diagnostic Engine V2", defaultValue: true,
+            serializations: new LocalUserProfileSerialization(LocalRegistryPath + "Use Diagnostic Engine V2"));
+
+        public static readonly Option<bool> CompilationEndCodeFix = new Option<bool>(nameof(InternalDiagnosticsOptions), "Enable Compilation End Code Fix", defaultValue: true,
+            serializations: new LocalUserProfileSerialization(LocalRegistryPath + "Enable Compilation End Code Fix"));
+
+        public static readonly Option<bool> UseCompilationEndCodeFixHeuristic = new Option<bool>(nameof(InternalDiagnosticsOptions), "Enable Compilation End Code Fix With Heuristic", defaultValue: true,
+            serializations: new LocalUserProfileSerialization(LocalRegistryPath + "Enable Compilation End Code Fix With Heuristic"));
+
+        public static readonly Option<bool> BuildErrorIsTheGod = new Option<bool>(nameof(InternalDiagnosticsOptions), "Make build errors to take over everything", defaultValue: false,
```

Comment:
ya, some of these internal option can now go away. I probably need to remove ones no longer needed. most of them are added to test user experience to see which one is better